utils/k8s/deprovision_deployment.yml

```diff
@@ -138,13 +157,14 @@
       when: storage_class is defined    
 
 - name: Delete the namespace.
-  shell: kubectl delete ns {{ app_ns }}
-  args:
-    executable: /bin/bash
+  k8s:
+    state: absent
+    kind: Namespace
+    name: "{{ app_ns }}"
 
 - name: Check if the PV is deleted.
-  shell: kubectl get pv {{ pv_name.stdout }}
-  args:
-    executable: /bin/bash
+  k8s_facts:
+    kind: PersistentVolume
+    name: "pv_name.resources.0.spec.volumName" 
   register: pv_result
-  failed_when: "pv_name.stdout in pv_result.stdout"
+  failed_when: "pv_result.resources | length > 1"
```

Comment:
can you test this when you have multiple volumes

Reply:
Fixed it for multiple volumes @gprasath by using label_selectors (field_selectors can also be  used)
Thanks for finding it!

Playbook code
```
- name: Check if the PV is deleted.
  k8s_facts:
    kind: PersistentVolume
    name: "{{ pvname }}"
    label_selectors:
      - "{{ app_label }}"
  register: pv_result
  failed_when: "pv_result.resources | length > 1"

```
Volumes Before deprovison :
```
[root@master-1558702621 deployers]# kubectl get pvc -npercona1
NAME                        STATUS    VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                    AGE
openebs-busybox-busybox-0   Bound     pvc-d93e2ecd-d97d-11e9-8eaa-0050569846e3   5G         RWO            openebs-sparse-sc-statefulset   36s
openebs-busybox-busybox-1   Bound     pvc-e73d9507-d97d-11e9-8eaa-0050569846e3   5G         RWO            openebs-sparse-sc-statefulset   13s
percona-mysql-claim         Bound     pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3   5G         RWO            openebs-sparse-sc-statefulset   2m

```
Volumes after Deprovision :
```
[root@master-1558702621 deployers]# kubectl get pvc -npercona1
NAME                        STATUS        VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                    AGE
openebs-busybox-busybox-0   Terminating   pvc-d93e2ecd-d97d-11e9-8eaa-0050569846e3   5G         RWO            openebs-sparse-sc-statefulset   5m
openebs-busybox-busybox-1   Terminating   pvc-e73d9507-d97d-11e9-8eaa-0050569846e3   5G         RWO            openebs-sparse-sc-statefulset   5m

```

Logs:

```
PLAY [localhost] ***************************************************************
2019-09-17T19:08:14.541691 (delta: 0.074049)         elapsed: 0.074049 ******** 
=============================================================================== 

TASK [Gathering Facts] *********************************************************
2019-09-17T19:08:14.561766 (delta: 0.020014)         elapsed: 0.094124 ******** 
ok: [127.0.0.1]

TASK [include_tasks] ***********************************************************
2019-09-17T19:08:24.422820 (delta: 9.861025)         elapsed: 9.955178 ******** 
included: /utils/fcm/create_testname.yml for 127.0.0.1

TASK [Record test instance/run ID] *********************************************
2019-09-17T19:08:24.538754 (delta: 0.11589)         elapsed: 10.071112 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Construct testname appended with runID] **********************************
2019-09-17T19:08:24.618122 (delta: 0.079312)         elapsed: 10.15048 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
2019-09-17T19:08:24.698535 (delta: 0.080361)         elapsed: 10.230893 ******* 
included: /utils/fcm/update_litmus_result_resource.yml for 127.0.0.1

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
2019-09-17T19:08:24.816517 (delta: 0.117911)         elapsed: 10.348875 ******* 
changed: [127.0.0.1] => {"changed": true, "checksum": "2fb43d87dfb4bba2e2fd205f2e69415962b85fac", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "8988f4be16232bf55e019321913ae47e", "mode": "0644", "owner": "root", "size": 419, "src": "/root/.ansible/tmp/ansible-tmp-1568747304.88-54560649298230/source", "state": "file", "uid": 0}

TASK [Analyze the cr yaml] *****************************************************
2019-09-17T19:08:25.708396 (delta: 0.891825)         elapsed: 11.240754 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "cat litmus-result.yaml", "delta": "0:00:01.150513", "end": "2019-09-17 19:08:27.344223", "rc": 0, "start": "2019-09-17 19:08:26.193710", "stderr": "", "stderr_lines": [], "stdout": "---\napiVersion: litmus.io/v1alpha1\nkind: LitmusResult\nmetadata:\n\n  # name of the litmus testcase\n  name: percona-deployment \nspec:\n\n  # holds information on the testcase\n  testMetadata:\n    app:  \n    chaostype:  \n\n  # holds the state of testcase,  manually updated by json merge patch\n  # result is the useful value today, but anticipate phase use in future \n  testStatus: \n    phase: in-progress  \n    result: none ", "stdout_lines": ["---", "apiVersion: litmus.io/v1alpha1", "kind: LitmusResult", "metadata:", "", "  # name of the litmus testcase", "  name: percona-deployment ", "spec:", "", "  # holds information on the testcase", "  testMetadata:", "    app:  ", "    chaostype:  ", "", "  # holds the state of testcase,  manually updated by json merge patch", "  # result is the useful value today, but anticipate phase use in future ", "  testStatus: ", "    phase: in-progress  ", "    result: none "]}

TASK [Apply the litmus result CR] **********************************************
2019-09-17T19:08:27.426418 (delta: 1.717967)         elapsed: 12.958776 ******* 
changed: [127.0.0.1] => {"changed": true, "failed_when_result": false, "method": "patch", "result": {"apiVersion": "litmus.io/v1alpha1", "kind": "LitmusResult", "metadata": {"clusterName": "", "creationTimestamp": "2019-09-17T17:42:12Z", "generation": 1, "name": "percona-deployment", "namespace": "", "resourceVersion": "4365779", "selfLink": "/apis/litmus.io/v1alpha1/litmusresults/percona-deployment", "uid": "7adf5230-d972-11e9-8eaa-0050569846e3"}, "spec": {"testMetadata": {}, "testStatus": {"phase": "in-progress", "result": "none"}}}}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
2019-09-17T19:08:28.774544 (delta: 1.348062)         elapsed: 14.306902 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Analyze the cr yaml] *****************************************************
2019-09-17T19:08:28.846811 (delta: 0.072165)         elapsed: 14.379169 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Apply the litmus result CR] **********************************************
2019-09-17T19:08:28.916805 (delta: 0.069931)         elapsed: 14.449163 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
2019-09-17T19:08:28.990467 (delta: 0.0736)         elapsed: 14.522825 ********* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
2019-09-17T19:08:29.069681 (delta: 0.079155)         elapsed: 14.602039 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
2019-09-17T19:08:29.155913 (delta: 0.086174)         elapsed: 14.688271 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
2019-09-17T19:08:29.241092 (delta: 0.085124)         elapsed: 14.77345 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
2019-09-17T19:08:29.334768 (delta: 0.093615)         elapsed: 14.867126 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [set_fact] ****************************************************************
2019-09-17T19:08:29.428393 (delta: 0.09356)         elapsed: 14.960751 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Deprovisioning the Application] ******************************************
2019-09-17T19:08:29.531079 (delta: 0.102621)         elapsed: 15.063437 ******* 
included: /utils/k8s/deprovision_deployment.yml for 127.0.0.1

TASK [Check if the application to be deleted is running.] **********************
2019-09-17T19:08:29.818321 (delta: 0.28717)         elapsed: 15.350679 ******** 
ok: [127.0.0.1] => {"attempts": 1, "changed": false, "resources": [{"apiVersion": "v1", "kind": "Pod", "metadata": {"annotations": {"openshift.io/scc": "restricted"}, "creationTimestamp": "2019-09-17T19:01:24Z", "generateName": "percona-fc7db745c-", "labels": {"name": "percona1", "openebs.io/target-affinity": "percona", "pod-template-hash": "973863017"}, "name": "percona-fc7db745c-wws4m", "namespace": "percona1", "ownerReferences": [{"apiVersion": "extensions/v1beta1", "blockOwnerDeletion": true, "controller": true, "kind": "ReplicaSet", "name": "percona-fc7db745c", "uid": "8b675f3c-d97d-11e9-8eaa-0050569846e3"}], "resourceVersion": "4365533", "selfLink": "/api/v1/namespaces/percona1/pods/percona-fc7db745c-wws4m", "uid": "8b819b15-d97d-11e9-8eaa-0050569846e3"}, "spec": {"containers": [{"args": ["--ignore-db-dir", "lost+found"], "env": [{"name": "MYSQL_ROOT_PASSWORD", "value": "k8sDem0"}], "image": "openebs/tests-custom-percona:latest", "imagePullPolicy": "IfNotPresent", "livenessProbe": {"exec": {"command": ["bash", "sql-test.sh"]}, "failureThreshold": 3, "initialDelaySeconds": 60, "periodSeconds": 1, "successThreshold": 1, "timeoutSeconds": 10}, "name": "percona", "ports": [{"containerPort": 3306, "name": "percona", "protocol": "TCP"}], "resources": {"limits": {"cpu": "500m"}, "requests": {"cpu": "500m"}}, "securityContext": {"capabilities": {"drop": ["KILL", "MKNOD", "SETGID", "SETUID"]}}, "terminationMessagePath": "/dev/termination-log", "terminationMessagePolicy": "File", "volumeMounts": [{"mountPath": "/var/lib/mysql", "name": "data-vol"}, {"mountPath": "/var/run/secrets/kubernetes.io/serviceaccount", "name": "default-token-rlt7d", "readOnly": true}]}], "dnsPolicy": "ClusterFirst", "imagePullSecrets": [{"name": "default-dockercfg-qwlhm"}], "nodeName": "node1-1558702621.mayalabs.io", "nodeSelector": {"node-role.kubernetes.io/compute": "true"}, "restartPolicy": "Always", "schedulerName": "default-scheduler", "securityContext": {"fsGroup": 1000450000, "seLinuxOptions": {"level": "s0:c21,c15"}}, "serviceAccount": "default", "serviceAccountName": "default", "terminationGracePeriodSeconds": 30, "tolerations": [{"effect": "NoSchedule", "key": "node.kubernetes.io/memory-pressure", "operator": "Exists"}], "volumes": [{"name": "data-vol", "persistentVolumeClaim": {"claimName": "percona-mysql-claim"}}, {"name": "default-token-rlt7d", "secret": {"defaultMode": 420, "secretName": "default-token-rlt7d"}}]}, "status": {"conditions": [{"lastProbeTime": null, "lastTransitionTime": "2019-09-17T19:01:25Z", "status": "True", "type": "Initialized"}, {"lastProbeTime": null, "lastTransitionTime": "2019-09-17T19:02:05Z", "message": "containers with unready status: [percona]", "reason": "ContainersNotReady", "status": "False", "type": "Ready"}, {"lastProbeTime": null, "lastTransitionTime": "2019-09-17T19:01:25Z", "status": "True", "type": "PodScheduled"}], "containerStatuses": [{"containerID": "docker://303e00eba5bd751b593d1fdf33731f0cc31bcd089a889314d73626dd5db02c34", "image": "docker.io/openebs/tests-custom-percona:latest", "imageID": "docker-pullable://docker.io/openebs/tests-custom-percona@sha256:17fa3de8c5cef79695960f25ee06b9de69ddf2bf50d553afa151076c3cd59e1e", "lastState": {"terminated": {"containerID": "docker://303e00eba5bd751b593d1fdf33731f0cc31bcd089a889314d73626dd5db02c34", "exitCode": 1, "finishedAt": "2019-09-17T19:07:32Z", "reason": "Error", "startedAt": "2019-09-17T19:07:32Z"}}, "name": "percona", "ready": false, "restartCount": 6, "state": {"waiting": {"message": "Back-off 5m0s restarting failed container=percona pod=percona-fc7db745c-wws4m_percona1(8b819b15-d97d-11e9-8eaa-0050569846e3)", "reason": "CrashLoopBackOff"}}}], "hostIP": "10.23.1.31", "phase": "Running", "podIP": "172.23.4.165", "qosClass": "Burstable", "startTime": "2019-09-17T19:01:25Z"}}]}

TASK [Obtaining the PVC name using application label.] *************************
2019-09-17T19:08:31.521994 (delta: 1.703599)         elapsed: 17.054352 ******* 
ok: [127.0.0.1] => {"changed": false, "resources": [{"apiVersion": "v1", "kind": "Pod", "metadata": {"annotations": {"openshift.io/scc": "restricted"}, "creationTimestamp": "2019-09-17T19:01:24Z", "generateName": "percona-fc7db745c-", "labels": {"name": "percona1", "openebs.io/target-affinity": "percona", "pod-template-hash": "973863017"}, "name": "percona-fc7db745c-wws4m", "namespace": "percona1", "ownerReferences": [{"apiVersion": "extensions/v1beta1", "blockOwnerDeletion": true, "controller": true, "kind": "ReplicaSet", "name": "percona-fc7db745c", "uid": "8b675f3c-d97d-11e9-8eaa-0050569846e3"}], "resourceVersion": "4365533", "selfLink": "/api/v1/namespaces/percona1/pods/percona-fc7db745c-wws4m", "uid": "8b819b15-d97d-11e9-8eaa-0050569846e3"}, "spec": {"containers": [{"args": ["--ignore-db-dir", "lost+found"], "env": [{"name": "MYSQL_ROOT_PASSWORD", "value": "k8sDem0"}], "image": "openebs/tests-custom-percona:latest", "imagePullPolicy": "IfNotPresent", "livenessProbe": {"exec": {"command": ["bash", "sql-test.sh"]}, "failureThreshold": 3, "initialDelaySeconds": 60, "periodSeconds": 1, "successThreshold": 1, "timeoutSeconds": 10}, "name": "percona", "ports": [{"containerPort": 3306, "name": "percona", "protocol": "TCP"}], "resources": {"limits": {"cpu": "500m"}, "requests": {"cpu": "500m"}}, "securityContext": {"capabilities": {"drop": ["KILL", "MKNOD", "SETGID", "SETUID"]}}, "terminationMessagePath": "/dev/termination-log", "terminationMessagePolicy": "File", "volumeMounts": [{"mountPath": "/var/lib/mysql", "name": "data-vol"}, {"mountPath": "/var/run/secrets/kubernetes.io/serviceaccount", "name": "default-token-rlt7d", "readOnly": true}]}], "dnsPolicy": "ClusterFirst", "imagePullSecrets": [{"name": "default-dockercfg-qwlhm"}], "nodeName": "node1-1558702621.mayalabs.io", "nodeSelector": {"node-role.kubernetes.io/compute": "true"}, "restartPolicy": "Always", "schedulerName": "default-scheduler", "securityContext": {"fsGroup": 1000450000, "seLinuxOptions": {"level": "s0:c21,c15"}}, "serviceAccount": "default", "serviceAccountName": "default", "terminationGracePeriodSeconds": 30, "tolerations": [{"effect": "NoSchedule", "key": "node.kubernetes.io/memory-pressure", "operator": "Exists"}], "volumes": [{"name": "data-vol", "persistentVolumeClaim": {"claimName": "percona-mysql-claim"}}, {"name": "default-token-rlt7d", "secret": {"defaultMode": 420, "secretName": "default-token-rlt7d"}}]}, "status": {"conditions": [{"lastProbeTime": null, "lastTransitionTime": "2019-09-17T19:01:25Z", "status": "True", "type": "Initialized"}, {"lastProbeTime": null, "lastTransitionTime": "2019-09-17T19:02:05Z", "message": "containers with unready status: [percona]", "reason": "ContainersNotReady", "status": "False", "type": "Ready"}, {"lastProbeTime": null, "lastTransitionTime": "2019-09-17T19:01:25Z", "status": "True", "type": "PodScheduled"}], "containerStatuses": [{"containerID": "docker://303e00eba5bd751b593d1fdf33731f0cc31bcd089a889314d73626dd5db02c34", "image": "docker.io/openebs/tests-custom-percona:latest", "imageID": "docker-pullable://docker.io/openebs/tests-custom-percona@sha256:17fa3de8c5cef79695960f25ee06b9de69ddf2bf50d553afa151076c3cd59e1e", "lastState": {"terminated": {"containerID": "docker://303e00eba5bd751b593d1fdf33731f0cc31bcd089a889314d73626dd5db02c34", "exitCode": 1, "finishedAt": "2019-09-17T19:07:32Z", "reason": "Error", "startedAt": "2019-09-17T19:07:32Z"}}, "name": "percona", "ready": false, "restartCount": 6, "state": {"waiting": {"message": "Back-off 5m0s restarting failed container=percona pod=percona-fc7db745c-wws4m_percona1(8b819b15-d97d-11e9-8eaa-0050569846e3)", "reason": "CrashLoopBackOff"}}}], "hostIP": "10.23.1.31", "phase": "Running", "podIP": "172.23.4.165", "qosClass": "Burstable", "startTime": "2019-09-17T19:01:25Z"}}]}

TASK [debug] *******************************************************************
2019-09-17T19:08:32.937131 (delta: 1.415038)         elapsed: 18.469489 ******* 
ok: [127.0.0.1] => (item={u'status': {u'podIP': u'172.23.4.165', u'startTime': u'2019-09-17T19:01:25Z', u'hostIP': u'10.23.1.31', u'phase': u'Running', u'conditions': [{u'status': u'True', u'type': u'Initialized', u'lastProbeTime': None, u'lastTransitionTime': u'2019-09-17T19:01:25Z'}, {u'status': u'False', u'reason': u'ContainersNotReady', u'lastProbeTime': None, u'type': u'Ready', u'message': u'containers with unready status: [percona]', u'lastTransitionTime': u'2019-09-17T19:02:05Z'}, {u'status': u'True', u'type': u'PodScheduled', u'lastProbeTime': None, u'lastTransitionTime': u'2019-09-17T19:01:25Z'}], u'qosClass': u'Burstable', u'containerStatuses': [{u'restartCount': 6, u'containerID': u'docker://303e00eba5bd751b593d1fdf33731f0cc31bcd089a889314d73626dd5db02c34', u'image': u'docker.io/openebs/tests-custom-percona:latest', u'imageID': u'docker-pullable://docker.io/openebs/tests-custom-percona@sha256:17fa3de8c5cef79695960f25ee06b9de69ddf2bf50d553afa151076c3cd59e1e', u'state': {u'waiting': {u'reason': u'CrashLoopBackOff', u'message': u'Back-off 5m0s restarting failed container=percona pod=percona-fc7db745c-wws4m_percona1(8b819b15-d97d-11e9-8eaa-0050569846e3)'}}, u'ready': False, u'lastState': {u'terminated': {u'startedAt': u'2019-09-17T19:07:32Z', u'reason': u'Error', u'finishedAt': u'2019-09-17T19:07:32Z', u'containerID': u'docker://303e00eba5bd751b593d1fdf33731f0cc31bcd089a889314d73626dd5db02c34', u'exitCode': 1}}, u'name': u'percona'}]}, u'kind': u'Pod', u'spec': {u'dnsPolicy': u'ClusterFirst', u'securityContext': {u'fsGroup': 1000450000, u'seLinuxOptions': {u'level': u's0:c21,c15'}}, u'nodeName': u'node1-1558702621.mayalabs.io', u'schedulerName': u'default-scheduler', u'serviceAccount': u'default', u'nodeSelector': {u'node-role.kubernetes.io/compute': u'true'}, u'terminationGracePeriodSeconds': 30, u'restartPolicy': u'Always', u'volumes': [{u'name': u'data-vol', u'persistentVolumeClaim': {u'claimName': u'percona-mysql-claim'}}, {u'secret': {u'defaultMode': 420, u'secretName': u'default-token-rlt7d'}, u'name': u'default-token-rlt7d'}], u'tolerations': [{u'operator': u'Exists', u'effect': u'NoSchedule', u'key': u'node.kubernetes.io/memory-pressure'}], u'imagePullSecrets': [{u'name': u'default-dockercfg-qwlhm'}], u'containers': [{u'livenessProbe': {u'timeoutSeconds': 10, u'exec': {u'command': [u'bash', u'sql-test.sh']}, u'successThreshold': 1, u'initialDelaySeconds': 60, u'periodSeconds': 1, u'failureThreshold': 3}, u'terminationMessagePath': u'/dev/termination-log', u'name': u'percona', u'image': u'openebs/tests-custom-percona:latest', u'args': [u'--ignore-db-dir', u'lost+found'], u'volumeMounts': [{u'mountPath': u'/var/lib/mysql', u'name': u'data-vol'}, {u'readOnly': True, u'mountPath': u'/var/run/secrets/kubernetes.io/serviceaccount', u'name': u'default-token-rlt7d'}], u'terminationMessagePolicy': u'File', u'env': [{u'name': u'MYSQL_ROOT_PASSWORD', u'value': u'k8sDem0'}], u'imagePullPolicy': u'IfNotPresent', u'securityContext': {u'capabilities': {u'drop': [u'KILL', u'MKNOD', u'SETGID', u'SETUID']}}, u'ports': [{u'protocol': u'TCP', u'name': u'percona', u'containerPort': 3306}], u'resources': {u'requests': {u'cpu': u'500m'}, u'limits': {u'cpu': u'500m'}}}], u'serviceAccountName': u'default'}, u'apiVersion': u'v1', u'metadata': {u'name': u'percona-fc7db745c-wws4m', u'labels': {u'pod-template-hash': u'973863017', u'name': u'percona1', u'openebs.io/target-affinity': u'percona'}, u'namespace': u'percona1', u'generateName': u'percona-fc7db745c-', u'resourceVersion': u'4365533', u'ownerReferences': [{u'kind': u'ReplicaSet', u'blockOwnerDeletion': True, u'name': u'percona-fc7db745c', u'controller': True, u'apiVersion': u'extensions/v1beta1', u'uid': u'8b675f3c-d97d-11e9-8eaa-0050569846e3'}], u'creationTimestamp': u'2019-09-17T19:01:24Z', u'annotations': {u'openshift.io/scc': u'restricted'}, u'selfLink': u'/api/v1/namespaces/percona1/pods/percona-fc7db745c-wws4m', u'uid': u'8b819b15-d97d-11e9-8eaa-0050569846e3'}}) => {
    "msg": "percona-mysql-claim"
}

TASK [Obtaining the PV name from PVC name.] ************************************
2019-09-17T19:08:33.108849 (delta: 0.171652)         elapsed: 18.641207 ******* 
ok: [127.0.0.1] => {"changed": false, "resources": [{"apiVersion": "v1", "kind": "PersistentVolumeClaim", "metadata": {"annotations": {"pv.kubernetes.io/bind-completed": "yes", "pv.kubernetes.io/bound-by-controller": "yes", "volume.beta.kubernetes.io/storage-provisioner": "openebs.io/provisioner-iscsi"}, "creationTimestamp": "2019-09-17T19:01:24Z", "finalizers": ["kubernetes.io/pvc-protection"], "labels": {"openebs.io/target-affinity": "percona"}, "name": "percona-mysql-claim", "namespace": "percona1", "resourceVersion": "4363762", "selfLink": "/api/v1/namespaces/percona1/persistentvolumeclaims/percona-mysql-claim", "uid": "8b68b1a1-d97d-11e9-8eaa-0050569846e3"}, "spec": {"accessModes": ["ReadWriteOnce"], "resources": {"requests": {"storage": "5G"}}, "storageClassName": "openebs-sparse-sc-statefulset", "volumeName": "pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3"}, "status": {"accessModes": ["ReadWriteOnce"], "capacity": {"storage": "5G"}, "phase": "Bound"}}]}

TASK [debug] *******************************************************************
2019-09-17T19:08:34.304536 (delta: 1.195626)         elapsed: 19.836894 ******* 
ok: [127.0.0.1] => (item={u'status': {u'phase': u'Bound', u'accessModes': [u'ReadWriteOnce'], u'capacity': {u'storage': u'5G'}}, u'kind': u'PersistentVolumeClaim', u'spec': {u'storageClassName': u'openebs-sparse-sc-statefulset', u'accessModes': [u'ReadWriteOnce'], u'resources': {u'requests': {u'storage': u'5G'}}, u'volumeName': u'pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3'}, u'apiVersion': u'v1', u'metadata': {u'name': u'percona-mysql-claim', u'labels': {u'openebs.io/target-affinity': u'percona'}, u'namespace': u'percona1', u'finalizers': [u'kubernetes.io/pvc-protection'], u'resourceVersion': u'4363762', u'creationTimestamp': u'2019-09-17T19:01:24Z', u'annotations': {u'pv.kubernetes.io/bound-by-controller': u'yes', u'volume.beta.kubernetes.io/storage-provisioner': u'openebs.io/provisioner-iscsi', u'pv.kubernetes.io/bind-completed': u'yes'}, u'selfLink': u'/api/v1/namespaces/percona1/persistentvolumeclaims/percona-mysql-claim', u'uid': u'8b68b1a1-d97d-11e9-8eaa-0050569846e3'}}) => {
    "msg": "pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3"
}

TASK [set_fact] ****************************************************************
2019-09-17T19:08:34.484427 (delta: 0.179835)         elapsed: 20.016785 ******* 
ok: [127.0.0.1] => {"ansible_facts": {"pvname": "pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3"}, "changed": false}

TASK [Check for presence & value of cas type annotation] ***********************
2019-09-17T19:08:34.598183 (delta: 0.113688)         elapsed: 20.130541 ******* 
ok: [127.0.0.1] => {"changed": false, "resources": [{"apiVersion": "v1", "kind": "PersistentVolume", "metadata": {"annotations": {"openEBSProvisionerIdentity": "node1-1558702621.mayalabs.io", "openebs.io/cas-type": "cstor", "pv.kubernetes.io/provisioned-by": "openebs.io/provisioner-iscsi"}, "creationTimestamp": "2019-09-17T19:01:25Z", "finalizers": ["kubernetes.io/pv-protection"], "labels": {"openebs.io/cas-type": "cstor", "openebs.io/storageclass": "openebs-sparse-sc-statefulset"}, "name": "pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3", "resourceVersion": "4363760", "selfLink": "/api/v1/persistentvolumes/pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3", "uid": "8be85e25-d97d-11e9-8eaa-0050569846e3"}, "spec": {"accessModes": ["ReadWriteOnce"], "capacity": {"storage": "5G"}, "claimRef": {"apiVersion": "v1", "kind": "PersistentVolumeClaim", "name": "percona-mysql-claim", "namespace": "percona1", "resourceVersion": "4363711", "uid": "8b68b1a1-d97d-11e9-8eaa-0050569846e3"}, "iscsi": {"fsType": "ext4", "iqn": "iqn.2016-09.com.openebs.cstor:pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3", "iscsiInterface": "default", "lun": 0, "targetPortal": "172.24.159.155:3260"}, "persistentVolumeReclaimPolicy": "Delete", "storageClassName": "openebs-sparse-sc-statefulset"}, "status": {"phase": "Bound"}}]}

TASK [debug] *******************************************************************
2019-09-17T19:08:35.887580 (delta: 1.289336)         elapsed: 21.419938 ******* 
ok: [127.0.0.1] => (item={u'status': {u'phase': u'Bound'}, u'kind': u'PersistentVolume', u'spec': {u'storageClassName': u'openebs-sparse-sc-statefulset', u'accessModes': [u'ReadWriteOnce'], u'claimRef': {u'kind': u'PersistentVolumeClaim', u'uid': u'8b68b1a1-d97d-11e9-8eaa-0050569846e3', u'resourceVersion': u'4363711', u'namespace': u'percona1', u'apiVersion': u'v1', u'name': u'percona-mysql-claim'}, u'capacity': {u'storage': u'5G'}, u'persistentVolumeReclaimPolicy': u'Delete', u'iscsi': {u'targetPortal': u'172.24.159.155:3260', u'iqn': u'iqn.2016-09.com.openebs.cstor:pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3', u'lun': 0, u'iscsiInterface': u'default', u'fsType': u'ext4'}}, u'apiVersion': u'v1', u'metadata': {u'name': u'pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3', u'labels': {u'openebs.io/cas-type': u'cstor', u'openebs.io/storageclass': u'openebs-sparse-sc-statefulset'}, u'finalizers': [u'kubernetes.io/pv-protection'], u'resourceVersion': u'4363760', u'creationTimestamp': u'2019-09-17T19:01:25Z', u'annotations': {u'pv.kubernetes.io/provisioned-by': u'openebs.io/provisioner-iscsi', u'openEBSProvisionerIdentity': u'node1-1558702621.mayalabs.io', u'openebs.io/cas-type': u'cstor'}, u'selfLink': u'/api/v1/persistentvolumes/pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3', u'uid': u'8be85e25-d97d-11e9-8eaa-0050569846e3'}}) => {
    "msg": "cstor"
}

TASK [Record the storage engine name] ******************************************
2019-09-17T19:08:36.079188 (delta: 0.191548)         elapsed: 21.611546 ******* 
ok: [127.0.0.1] => (item={u'status': {u'phase': u'Bound'}, u'kind': u'PersistentVolume', u'spec': {u'storageClassName': u'openebs-sparse-sc-statefulset', u'accessModes': [u'ReadWriteOnce'], u'claimRef': {u'kind': u'PersistentVolumeClaim', u'uid': u'8b68b1a1-d97d-11e9-8eaa-0050569846e3', u'resourceVersion': u'4363711', u'namespace': u'percona1', u'apiVersion': u'v1', u'name': u'percona-mysql-claim'}, u'capacity': {u'storage': u'5G'}, u'persistentVolumeReclaimPolicy': u'Delete', u'iscsi': {u'targetPortal': u'172.24.159.155:3260', u'iqn': u'iqn.2016-09.com.openebs.cstor:pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3', u'lun': 0, u'iscsiInterface': u'default', u'fsType': u'ext4'}}, u'apiVersion': u'v1', u'metadata': {u'name': u'pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3', u'labels': {u'openebs.io/cas-type': u'cstor', u'openebs.io/storageclass': u'openebs-sparse-sc-statefulset'}, u'finalizers': [u'kubernetes.io/pv-protection'], u'resourceVersion': u'4363760', u'creationTimestamp': u'2019-09-17T19:01:25Z', u'annotations': {u'pv.kubernetes.io/provisioned-by': u'openebs.io/provisioner-iscsi', u'openEBSProvisionerIdentity': u'node1-1558702621.mayalabs.io', u'openebs.io/cas-type': u'cstor'}, u'selfLink': u'/api/v1/persistentvolumes/pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3', u'uid': u'8be85e25-d97d-11e9-8eaa-0050569846e3'}}) => {"ansible_facts": {"stg_engine": "cstor"}, "changed": false, "item": {"apiVersion": "v1", "kind": "PersistentVolume", "metadata": {"annotations": {"openEBSProvisionerIdentity": "node1-1558702621.mayalabs.io", "openebs.io/cas-type": "cstor", "pv.kubernetes.io/provisioned-by": "openebs.io/provisioner-iscsi"}, "creationTimestamp": "2019-09-17T19:01:25Z", "finalizers": ["kubernetes.io/pv-protection"], "labels": {"openebs.io/cas-type": "cstor", "openebs.io/storageclass": "openebs-sparse-sc-statefulset"}, "name": "pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3", "resourceVersion": "4363760", "selfLink": "/api/v1/persistentvolumes/pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3", "uid": "8be85e25-d97d-11e9-8eaa-0050569846e3"}, "spec": {"accessModes": ["ReadWriteOnce"], "capacity": {"storage": "5G"}, "claimRef": {"apiVersion": "v1", "kind": "PersistentVolumeClaim", "name": "percona-mysql-claim", "namespace": "percona1", "resourceVersion": "4363711", "uid": "8b68b1a1-d97d-11e9-8eaa-0050569846e3"}, "iscsi": {"fsType": "ext4", "iqn": "iqn.2016-09.com.openebs.cstor:pvc-8b68b1a1-d97d-11e9-8eaa-0050569846e3", "iscsiInterface": "default", "lun": 0, "targetPortal": "172.24.159.155:3260"}, "persistentVolumeReclaimPolicy": "Delete", "storageClassName": "openebs-sparse-sc-statefulset"}, "status": {"phase": "Bound"}}}

TASK [Replace the PVC name in application deployer spec.] **********************
2019-09-17T19:08:36.292949 (delta: 0.213691)         elapsed: 21.825307 ******* 
changed: [127.0.0.1] => {"changed": true, "msg": "2 replacements made"}

TASK [Replace the storageclass placeholder with provider] **********************
2019-09-17T19:08:36.785122 (delta: 0.492127)         elapsed: 22.31748 ******** 
changed: [127.0.0.1] => {"changed": true, "msg": "1 replacements made"}

TASK [Get the application label values from env] *******************************
2019-09-17T19:08:37.124540 (delta: 0.338401)         elapsed: 22.656898 ******* 
ok: [127.0.0.1] => {"ansible_facts": {"app_lkey": "name", "app_lvalue": "percona1"}, "changed": false}

TASK [Replace the application label placeholder] *******************************
2019-09-17T19:08:37.252565 (delta: 0.127954)         elapsed: 22.784923 ******* 
changed: [127.0.0.1] => {"changed": true, "msg": "5 replacements made"}

TASK [Delete the application deployment.] **************************************
2019-09-17T19:08:37.641759 (delta: 0.389123)         elapsed: 23.174117 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl delete -f percona.yml -n percona1", "delta": "0:00:01.881859", "end": "2019-09-17 19:08:39.824093", "rc": 0, "start": "2019-09-17 19:08:37.942234", "stderr": "", "stderr_lines": [], "stdout": "deployment.apps \"percona\" deleted\npersistentvolumeclaim \"percona-mysql-claim\" deleted\nservice \"percona-mysql\" deleted", "stdout_lines": ["deployment.apps \"percona\" deleted", "persistentvolumeclaim \"percona-mysql-claim\" deleted", "service \"percona-mysql\" deleted"]}

TASK [Check if the PVC is deleted.] ********************************************
2019-09-17T19:08:39.913712 (delta: 2.271855)         elapsed: 25.44607 ******** 
ok: [127.0.0.1] => {"attempts": 1, "changed": false, "resources": []}

TASK [Check if the ctrl-pod is deleted.] ***************************************
2019-09-17T19:08:41.099565 (delta: 1.185788)         elapsed: 26.631923 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Check if the  replica-pod is deleted.] ***********************************
2019-09-17T19:08:41.191189 (delta: 0.091574)         elapsed: 26.723547 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Check if the target pod is deleted.] *************************************
2019-09-17T19:08:41.292467 (delta: 0.101228)         elapsed: 26.824825 ******* 
ok: [127.0.0.1] => {"attempts": 1, "changed": false, "resources": []}

TASK [Check status of the pool deployments from cvr] ***************************
2019-09-17T19:08:42.528136 (delta: 1.23559)         elapsed: 28.060494 ******** 
ok: [127.0.0.1] => {"attempts": 1, "changed": false, "resources": []}

TASK [Delete the namespace.] ***************************************************
2019-09-17T19:08:43.844505 (delta: 1.3163)         elapsed: 29.376863 ********* 
changed: [127.0.0.1] => {"changed": true, "method": "delete", "result": {"apiVersion": "v1", "kind": "Namespace", "metadata": {"annotations": {"openshift.io/sa.scc.mcs": "s0:c21,c15", "openshift.io/sa.scc.supplemental-groups": "1000450000/10000", "openshift.io/sa.scc.uid-range": "1000450000/10000"}, "creationTimestamp": "2019-09-17T19:01:03Z", "deletionTimestamp": "2019-09-17T19:08:44Z", "name": "percona1", "resourceVersion": "4365882", "selfLink": "/api/v1/namespaces/percona1", "uid": "7f23e40a-d97d-11e9-8eaa-0050569846e3"}, "spec": {"finalizers": ["kubernetes"]}, "status": {"phase": "Terminating"}}}

TASK [Check if the PV is deleted.] *********************************************
2019-09-17T19:08:45.000967 (delta: 1.156403)         elapsed: 30.533325 ******* 
ok: [127.0.0.1] => {"changed": false, "failed_when_result": false, "resources": []}

TASK [set_fact] ****************************************************************
2019-09-17T19:08:46.416623 (delta: 1.415595)         elapsed: 31.948981 ******* 
ok: [127.0.0.1] => {"ansible_facts": {"flag": "Pass"}, "changed": false}

TASK [include_tasks] ***********************************************************
2019-09-17T19:08:46.528363 (delta: 0.111684)         elapsed: 32.060721 ******* 
included: /utils/fcm/update_litmus_result_resource.yml for 127.0.0.1

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
2019-09-17T19:08:46.642537 (delta: 0.114098)         elapsed: 32.174895 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Analyze the cr yaml] *****************************************************
2019-09-17T19:08:46.709313 (delta: 0.066717)         elapsed: 32.241671 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Apply the litmus result CR] **********************************************
2019-09-17T19:08:46.776774 (delta: 0.06739)         elapsed: 32.309132 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
2019-09-17T19:08:46.845008 (delta: 0.068181)         elapsed: 32.377366 ******* 
changed: [127.0.0.1] => {"changed": true, "checksum": "6bdd1ff2d018e616e50bff842ad05cafa7ec7f38", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "5d64f6c678b53bfda23b3b57fa0d82ad", "mode": "0644", "owner": "root", "size": 417, "src": "/root/.ansible/tmp/ansible-tmp-1568747326.92-165391233425911/source", "state": "file", "uid": 0}

TASK [Analyze the cr yaml] *****************************************************
2019-09-17T19:08:49.540506 (delta: 2.695438)         elapsed: 35.072864 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "cat litmus-result.yaml", "delta": "0:00:01.168498", "end": "2019-09-17 19:08:51.000305", "rc": 0, "start": "2019-09-17 19:08:49.831807", "stderr": "", "stderr_lines": [], "stdout": "---\napiVersion: litmus.io/v1alpha1\nkind: LitmusResult\nmetadata:\n\n  # name of the litmus testcase\n  name: percona-deployment \nspec:\n\n  # holds information on the testcase\n  testMetadata:\n    app:  \n    chaostype:  \n\n  # holds the state of testcase,  manually updated by json merge patch\n  # result is the useful value today, but anticipate phase use in future \n  testStatus: \n    phase: completed  \n    result: Pass ", "stdout_lines": ["---", "apiVersion: litmus.io/v1alpha1", "kind: LitmusResult", "metadata:", "", "  # name of the litmus testcase", "  name: percona-deployment ", "spec:", "", "  # holds information on the testcase", "  testMetadata:", "    app:  ", "    chaostype:  ", "", "  # holds the state of testcase,  manually updated by json merge patch", "  # result is the useful value today, but anticipate phase use in future ", "  testStatus: ", "    phase: completed  ", "    result: Pass "]}

TASK [Apply the litmus result CR] **********************************************
2019-09-17T19:08:51.071158 (delta: 1.530581)         elapsed: 36.603516 ******* 
changed: [127.0.0.1] => {"changed": true, "failed_when_result": false, "method": "patch", "result": {"apiVersion": "litmus.io/v1alpha1", "kind": "LitmusResult", "metadata": {"clusterName": "", "creationTimestamp": "2019-09-17T17:42:12Z", "generation": 1, "name": "percona-deployment", "namespace": "", "resourceVersion": "4365982", "selfLink": "/apis/litmus.io/v1alpha1/litmusresults/percona-deployment", "uid": "7adf5230-d972-11e9-8eaa-0050569846e3"}, "spec": {"testMetadata": {}, "testStatus": {"phase": "completed", "result": "Pass"}}}}

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=31   changed=11   unreachable=0    failed=0   

2019-09-17T19:08:52.348682 (delta: 1.277388)         elapsed: 37.88104 ******** 
=============================================================================== 

```